src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs

```diff
@@ -3848,24 +3847,22 @@ private void WriteElements(string source, string? enumSource, ElementAccessor[]
 
         private void WriteText(string source, TextAccessor text)
         {
-            if (text.Mapping is PrimitiveMapping)
+            if (text.Mapping is PrimitiveMapping primitive)
             {
-                PrimitiveMapping mapping = (PrimitiveMapping)text.Mapping;
                 Writer.Write("WriteValue(");
-                if (text.Mapping is EnumMapping)
+                if (text.Mapping is EnumMapping enum)
```

Comment:
Sorry wrong suggestion
```suggestion
                if (text.Mapping is EnumMapping enumMapping)
```